src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -84,7 +84,7 @@ public static ExpressionSyntax Parenthesize(this ExpressionSyntax expression, bo
             int position,
             SemanticModel semanticModel)
         {
-            if (targetType.ContainsAnonymousType())
+            if (targetType.ContainsAnonymousType() || expression.Kind() == SyntaxKind.CastExpression)
```

Comment:
This syntactic check is wrong. Note that this means that this method would stop adding casts to *any* type. For example, it would no longer be able to cast `(B)c` to `(A)(B)c`.